### ISSUE TITLE
Implemented size, addSelect and memberOf DQL functions

### ISSUE THREAD

**Issue by patrickvale** [Happyr/Doctrine-Specification#163]
Hi, this is a great project - thanks for all the hard work in making it!

I'm using it in production, and found a need to add support for DQL member of, size and addSelect functions. So I [forked](https://github.com/dailyinfo/Doctrine-Specification/tree/dql-functions) and implemented those. 

Before submitting a PR I wanted to check that these are features of interest, and that the way they've been implemented is OK.

The comparison order (left/right operator) is effectively flipped for a member of comparison (generating 'value MEMBER OF dqlalias.field' rather than for example 'dqlalias.field LIKE value'). I've implemented the check for this in a slightly hacky way within the Filter/Comparison class. I'd be open to suggestions as to how to do it better, but didn't want to refactor the base Comparison class without reaching out first.

Again, thanks for this great project, and hopefully these features may be of use to some other people - they are certainly making my life easier!

**Comment by edefimov** [comment]
The `AddSelect` feature is included in #156

**Comment by peter-gribanov** [comment]
Now you can using `select` and `addSelect` in specification:
```php
$spec = Spec::andX(
    Spec::select('field_name1'),
    Spec::addSelect('field_name2')
)
```

**Comment by peter-gribanov** [comment]
Also you can use Doctrine functions in specification:
```php
$spec = Spec::gt(Spec::SIZE('my_field'), 0);
```

**Comment by peter-gribanov** [comment]
Solved by #172, #175 and #186
